showcase/tests/integration/components/hds/app-side-nav/index-test.js

```diff
@@ -0,0 +1,350 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import {
+  render,
+  click,
+  resetOnerror,
+  settled,
+  triggerKeyEvent,
+} from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+class MockEventTarget extends EventTarget {}
+
+module('Integration | Component | hds/app-side-nav/index', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    // Mock window.matchMedia to control media query events
+    let mockMedia = new MockEventTarget();
+    mockMedia.matches = true;
+
+    this.__matchMedia = window.matchMedia;
+
+    this.mockMedia = () => {
+      window.matchMedia = () => mockMedia;
+    };
+
+    this.changeBrowserSize = async (isDesktop) => {
+      mockMedia.matches = isDesktop;
+      mockMedia.dispatchEvent(
+        new MediaQueryListEvent('change', {
+          matches: isDesktop,
+        })
+      );
+      await settled();
+    };
+  });
+
+  hooks.afterEach(function () {
+    resetOnerror();
+    window.matchMedia = this.__matchMedia;
+  });
+
+  test('it should render the component with a CSS class that matches the component name', async function (assert) {
+    await render(
+      hbs`<Hds::AppSideNav id="test-app-side-nav" @hasA11yRefocus={{false}} />`
+    );
+    assert.dom('#test-app-side-nav').hasClass('hds-app-side-nav');
+  });
+
+  // CONTENT
```

Comment:
Thought: I know we haven't really done this with other components, but I think it would be beneficial to add a gut check test to ensure keyboard focus is working correctly. Like when the sidenav is collapsed, you can't tab to the contents.

Reply:
On line 183 there's already a test for this:
```
assert.dom('.hds-app-side-nav-hide-when-minimized').hasAttribute('inert');
```

The framed example wasn't preventing tabbing as it should previously (before I fixed it) because I had missed updating the class names in the file (The classes have to be added manually to Lists in the AppSideNav content as in the SideNav).